packages/grid/test/column-reordering.common.js

```diff
@@ -248,6 +248,32 @@ describe('reordering simple grid', () => {
       expectVisualOrder(grid, [2, 1]);
     });
 
+    it('should reorder focus button mode columns', () => {
+      grid = fixtureSync(`
+        <vaadin-grid column-reordering-allowed>
+          <vaadin-grid-column path="name"></vaadin-grid-column>
+          <vaadin-grid-column path="age"></vaadin-grid-column>
+        </vaadin-grid>
+      `);
+      grid.querySelectorAll('vaadin-grid-column').forEach((col) => {
+        col._focusButtonMode = true;
+      });
+      grid.items = [
+        { name: 'John', age: 30 },
+        { name: 'Ringo', age: 40 },
+      ];
+      flushGrid(grid);
+
+      const headerContent = [
+        getContainerCell(grid.$.header, 0, 0)._content,
+        getContainerCell(grid.$.header, 0, 1)._content,
+      ];
+      dragAndDropOver(headerContent[0], headerContent[1]);
+
+      expect(getVisualHeaderCellContent(grid, 0, 0).innerText).to.be.equal('Age');
```

Comment:
This test currently passes even without the PR changes. It likely needs an update after https://github.com/vaadin/web-components/pull/7274 has been merged.

Reply:
Rebased, added a test for the body cells, and still kept the header one as insurance.